apps/studio/src/components/tableinfo/TableSchema.vue

```diff
@@ -366,23 +356,21 @@ export default Vue.extend({
       }
     },
     // submission methods
-    async submitApply(): Promise<void> {
-      try {
-        this.error = null
-        const changes = this.collectChanges()
-        await this.connection.alterTable(changes)
-
-        this.clearChanges()
-        await this.$store.dispatch('updateTableColumns', this.table)
-        this.$nextTick(() => {
+    async submitApply() {
+      this.error = null
+      this.trigger(AppEvent.alterTable, {
```

Comment:
Why did you move the alterTable function to CoreTabs? I don't see why it can't stay in here?

We should not be passing functions through the event bus